src/Illuminate/Filesystem/Filesystem.php

```diff
@@ -602,6 +603,25 @@ public function allFiles($directory, $hidden = false)
         );
     }
 
+    /**
+     * Get the PSR-4 class name for a file.
+     *
+     * @return ($files is string ? class-string : array<int, class-string>)
+     */
+    public function classFromFile(array|string $files, ?string $basePath = null): string|array
+    {
+        $namespaces = app('composer')->getNamespaces();
```

Comment:
This makes this method only valid from within Laravel application and will break when used outside of Laravel.

Reply:
What parts of Laravel are used outside of Laravel? I can move this method to the Composer class so everything is self contained (and create a Composer facade so you can simply use `Composer::classFromFile()`, but the functionality still relies on `base_path` which wouldn't necessarily defined outside of Laravel?